src/SprykerEco/Zed/ArvatoRss/Communication/Plugin/Oms/Condition/IsStoreOrderSuccessfulPlugin.php

```diff
@@ -43,36 +44,36 @@ public function check(SpySalesOrderItem $orderItem)
      */
     protected function isStoreOrderSuccessful($orderReference)
     {
-        $storeOrderApiCallLog = $this->getApiCallLogEntry($orderReference);
-        if ($storeOrderApiCallLog === null) {
+        $storeOrderApiCallLogTransfer = $this->getApiCallLogEntry($orderReference);
+
+        if ($storeOrderApiCallLogTransfer === null) {
             return false;
         }
 
-        return $this->isTransactionSuccessfull($storeOrderApiCallLog);
+        return $this->isTransactionSuccessful($storeOrderApiCallLogTransfer);
     }
 
     /**
      * @param string $orderReference
      *
-     * @return \Orm\Zed\ArvatoRss\Persistence\SpyArvatoRssApiCallLog
+     * @return \Generated\Shared\Transfer\ArvatoRssApiCallLogTransfer
      */
-    protected function getApiCallLogEntry($orderReference)
+    protected function getApiCallLogEntry(string $orderReference): ArvatoRssApiCallLogTransfer
```

Comment:
We do not prefer business logic in plugins.

Reply:
I agree but I'm not sure we have time for creating business model right now.